office365/sharepoint/actions/upload_session.py

```diff
@@ -66,23 +67,23 @@ def _build_upload_session_query(self, response):
         :type response: requests.Response
         """
         st = os.stat(self._source_path)
-        # upload a file in chunks
-        f_pos = 0
-        fh = open(self._source_path, 'rb')
-        for piece in read_in_chunks(fh, size=self._chunk_size):
-            if f_pos == 0:
-                upload_result = self.file.start_upload(self._upload_id, piece)
-                self._upload_results.append(upload_result)
-            elif f_pos + len(piece) < st.st_size:
-                upload_result = self.file.continue_upload(self._upload_id, f_pos, piece)
-                self._upload_results.append(upload_result)
```

Comment:
@vgrem 
also I am questioning the need to store the result. It is not used anywhere but consumes memory as well
or am I missing the place where it is utilized?

Reply:
Greetings! First of all, that you for catching it! Agree with you, the way how it is implemented in the current version, does not prevent the whole file to be read into memory.  
That basically contradicts to the advantage of utilizing lazy load approach which was supposed to bring an efficiency in terms of memory consumption.

And last but not least, thank you for the proposal and PR, lets merge it and see if works as expected.

Reply:
speaking of storing of results, right call! :+1:

Reply:
great, thank you!